spec/features/consumer/shopping/shopping_spec.rb

```diff
@@ -159,6 +159,31 @@
             end
           end
         end
+
+        describe "two order cycles and more than 20 products for each" do
+          before do
+            20.times do
+              product = create(:simple_product, supplier: supplier)
+              add_variant_to_order_cycle(exchange1, product.variants.first)
+              add_variant_to_order_cycle(exchange2, product.variants.first)
+            end
```

Comment:
this is going to be quite expensive. We could find a way to stub the number of items per page so we test the same but with 2 products.

Reply:
I can easily change the value of the `DEFAULT_PER_PAGE` constant inside `ProductsRenderer` which is in charge of doing the jsonification of products with this line:
```ruby
stub_const("#{ProductsRenderer}::DEFAULT_PER_PAGE", 1)
```

But, inside the angularjs controller, to test if it should load more products, the conditional testing is: 
```js
if ($scope.page * $scope.per_page) <= Products.products.length
      $scope.loadMoreProducts()
```

and `$scope.per_page` in initialized with a simple: `$scope.per_page = 10`

Which also means that `ProductsRenderer::DEFAULT_PER_PAGE` and `ProductsCtrl.$scop.per_page` must be the same to work well.

Reply:
So I'm looking for a way to change the value of an Angularjs controller constant in a Capybara feature test...

Reply:
In this case the cost is ~10 seconds on the build, but the reward is confidence that products are loading correctly in the shopfront. 

Maybe we can look at improving the efficiency in a separate tech-debt issue? It would be good to get this in the release...

Reply:
Yes, I think you're right : it probably needs some none-negligible time to create such a test, but we should merge this PR because it solves a lot of performing issues (at least unnecessary requests loads) in the app.
So: 
 - Happy to create a tech debt issue 
 - Let's merge it (if everyone is ok with the solution/code review of course)

Reply:
Issue created: #7508